src/pages/dashboard/panels/CollectionStats.tsx

```diff
@@ -1,23 +1,50 @@
 import React from 'react';
-import { useSelector } from 'react-redux';
-import { Link } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link, useNavigate } from 'react-router-dom';
 import prettyBytes from 'pretty-bytes';
 
 import ShokoPanel from '@/components/Panels/ShokoPanel';
 import { useDashbordStatsQuery } from '@/core/react-query/dashboard/queries';
+import { resetFilter } from '@/core/slices/collection';
+import { addFilterCriteriaToStore } from '@/core/utilities/filter';
+import useEventCallback from '@/hooks/useEventCallback';
 
 import type { RootState } from '@/core/store';
 
 const Item = (
-  { link, title, value = 0 }: { title: string, value?: string | number, link?: string },
-) => (
-  <div className="flex">
-    <div className="grow">
-      {title}
+  { filter, link, title, value = 0 }: { title: string, value?: string | number, link?: string, filter?: string },
+) => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const handleMissingFilter = useEventCallback((filterName: string) => {
+    dispatch(resetFilter());
+    addFilterCriteriaToStore(filterName).then(() => {
+      navigate('/webui/collection');
+    }).catch(console.error);
+  });
+
+  return (
+    /* eslint-disable no-nested-ternary */
+    <div className="flex">
+      <div className="grow">
+        {title}
+      </div>
+      {link
+        ? <Link to={link} className="font-semibold text-panel-text-primary">{value}</Link>
+        : filter
+        ? (
+          <div
+            className="cursor-pointer font-semibold text-panel-text-primary"
+            onClick={() => handleMissingFilter(filter)}
+          >
+            {value}
+          </div>
+        )
+        : <div>{value}</div>}
     </div>
-    {link ? <Link to={link} className="font-semibold text-panel-text-primary">{value}</Link> : <div>{value}</div>}
-  </div>
-);
+    /* eslint-enable no-nested-ternary */
```

Comment:
```suggestion
```

Reply:
It will make sense when you look at the other suggestion I made for the code at the top 😔